pkgs/development/python-modules/playwright/default.nix

```diff
@@ -0,0 +1,223 @@
+        'SCRIPT_PATH="$(cd "$(dirname "$0")" ; pwd -P)"' \
+        "SCRIPT_PATH=$out"
+
+      patchShebangs playwright.sh
+      patchShebangs package/bin/*.sh
+    '';
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir -p $out/bin
+      mv playwright.sh package $out/
+
+      # The actual driver script needs to stay at $out/playwright.sh because the Python bindings
+      # expect it to be there. But we can add a symlink so it's available in PATH as well.
+      ln -s $out/playwright.sh $out/bin/playwright-driver
+    '';
+
+    passthru = {
+      inherit filename;
+    };
+  };
+
+  browsers-mac = stdenv.mkDerivation {
+    pname = "playwright-browsers";
+    version = driverVersion;
+
+    src = runCommand "playwright-browsers-base" {
+      outputHashMode = "recursive";
+      outputHashAlgo = "sha256";
+      outputHash = {
+        x86_64-darwin = "1nd6ll6dj0aqic3hh65y0br051ckbcap8spqbvzm19g8dn7vgv7m";
+      }.${system};
+    } ''
+      export PLAYWRIGHT_BROWSERS_PATH=$out
+      ${driver}/playwright.sh install
+      rm -r $out/.links
+    '';
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir $out
+      cp -r * $out/
+    '';
+
+    doCheck = false;
+  };
+
+  browsers-linux = let
+    fontconfig = makeFontsConf {
+      fontDirectories = [];
+    };
+  in runCommand "playwright-browsers" {
+    nativeBuildInputs = [ makeWrapper ];
+  } ''
+    mkdir -p $out/chromium-1005/chrome-linux $out/ffmpeg-1007
+
+    # See here for the Chrome options:
+    # https://github.com/NixOS/nixpkgs/issues/136207#issuecomment-908637738
+    makeWrapper ${chromium}/bin/chromium $out/chromium-1005/chrome-linux/chrome \
+      --set SSL_CERT_FILE ${cacert}/etc/ssl/certs/ca-bundle.crt \
+      --set FONTCONFIG_FILE ${fontconfig}
+
+    ln -s ${ffmpeg}/bin/ffmpeg $out/ffmpeg-1007/ffmpeg-linux
+  '';
+in
+buildPythonPackage rec {
+  pname = "playwright";
+  version = "1.25.1";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    sha256 = "sha256-TWc/fzRl5w4+kMF8odJfFwt+JSpCfyK4BJNB2mq8oTM=";
+  };
+
+  propagatedBuildInputs = [
+    autobahn
+    greenlet
+    objgraph
+    pillow
+    pixelmatch
+    pyee
+    pyopenssl
+    requests
+    service-identity
+    websockets
+  ];
+
+  format = "pyproject";
+
+  postPatch = ''
+    # The package seems to work with pyee 9 as well, which is what nixpkgs has packaged.
+    substituteInPlace meta.yaml --replace "- pyee ==8.1.0" ""
+    substituteInPlace setup.py \
+        --replace '"pyee==8.1.0",' "" \
+        --replace '"setuptools-scm==6.3.2",' "" \
+        --replace '"websockets==10.1",' ""
+
+    # Replace the functionality that setuptools-scm provides by adding the version to a file.
+    echo 'version = "${version}"' > playwright/_repo_version.py
+  '';
+
+  preBuild = ''
+    mkdir -p driver
+    cp ${driver.src} ./driver/${driver.filename}
```

Comment:
Can we symlink here?

Reply:
On second though, we can skip that entirely. It's only to fool setup.py anyway, so it's probably smarter to just patch the setup script so that it won't try and extract the driver